hw/xbox/mcpx_apu.c

```diff
@@ -461,7 +465,34 @@ static uint64_t gp_read(void *opaque,
 {
     MCPXAPUState *d = opaque;
 
-    uint64_t r = d->gp.regs[addr];
+    assert(size == 4);
```

Comment:
Let's be safe here and also `assert` that the access is aligned... e.g.:
```
assert(addr & 3 == 0)
```

Reply:
Will do, but this will delay the PR as we have to test at least a handful of games.

Reply:
Please and thank you!